src/Adapter/Form/ChoiceProvider/TaxRuleGroupChoiceProvider.php

```diff
@@ -53,26 +75,32 @@ public function getChoices()
      */
     public function getChoicesAttributes(): array
     {
-        $attrs = [];
+        $address = new Address();
+        $address->id_country = $this->countryId;
+
+        $taxRates = [];
         foreach ($this->getRules() as $rule) {
-            // Keep first one found
-            if (!empty($attrs[$rule['name']]['data-tax-rate'])) {
-                continue;
-            }
+            $taxRulesGroupId = new TaxRulesGroupId((int) $rule['id_tax_rules_group']);
+            $stateId = $this->taxRulesGroupRepository->getTaxRulesGroupDefaultStateId($taxRulesGroupId, new CountryId($this->countryId));
+            $address->id_state = $stateId;
 
-            $attrs[$rule['name']] = [
-                'data-tax-rate' => !empty($rule['rate']) ? $rule['rate'] : 0,
+            $taxManagerFactory = TaxManagerFactory::getManager($address, $taxRulesGroupId->getValue());
+            $taxRate = $taxManagerFactory->getTaxCalculator()->getTotalRate();
```

Comment:
```suggestion
            $taxRate = $this->taxComputer->getTaxRateByState($taxRulesGroupId, new CountryId($this->countryId), new State($stateId));
```

Reply:
I you implement the `getTaxRateByState` as I suggested in TaxComputer comments, the this service doesn't need to reimplement this logic it is all handled in TaxComputer